flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java

```diff
@@ -86,8 +86,10 @@
  * arrived after the checkpoint from which we restore.
  *
  * @param <IN> Type of the elements emitted by this sink
+ * @deprecated Use {@link org.apache.flink.connector.file.sink.FileSink} instead.
  */
 @PublicEvolving
+@Deprecated
```

Comment:
In python class `StreamingFileSink`, we can also add the deprecated note.

Reply:
Thanks, I'll add it there too!